wrappers/golang/hash/keccak/hasher.go

```diff
@@ -0,0 +1,70 @@
+package keccak
+
+// #cgo CFLAGS: -I./include/
+// #include "keccak.h"
+import "C"
+
+import (
+	"unsafe"
+
+	core "github.com/ingonyama-zk/icicle/v2/wrappers/golang/core"
+	cr "github.com/ingonyama-zk/icicle/v2/wrappers/golang/cuda_runtime"
+)
+
+type KeccakConfig struct {
+	Ctx                cr.DeviceContext
+	areInputsOnDevice  bool
+	areOutputsOnDevice bool
+	IsAsync            bool
+}
+
+func GetDefaultKeccakConfig() KeccakConfig {
+	ctx, _ := cr.GetDefaultDeviceContext()
+	return KeccakConfig{
+		ctx,
+		false,
+		false,
+		false,
+	}
+}
+
+func KeccakCheck(input core.HostOrDeviceSlice, output core.HostOrDeviceSlice, cfg *KeccakConfig) (unsafe.Pointer, unsafe.Pointer, unsafe.Pointer) {
```

Comment:
This can be a private method since it is only used by Keccak256 and Keccak512
```suggestion
func keccakCheck(input core.HostOrDeviceSlice, output core.HostOrDeviceSlice, cfg *KeccakConfig) (unsafe.Pointer, unsafe.Pointer, unsafe.Pointer) {
```

Also change the calls in Keccak512 and Keccak256